lib/enum/basic_enum.rb

```diff
@@ -85,66 +98,79 @@ def element_frequency_count
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-find-all
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-select
   def elements_ending_in_er
+    collection.find_all {|x| x.include?("er")}
   end
 
 
   # Refer to find_index
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-find-index
   def index_of_first_awesome_element
+    collection.find_index {|x| x[:awesome] }
   end
 
   # Refer to group_by
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-group-by
   def group_elements_by_favorite_language
+    collection.group_by { |fido| fido.fetch(:favorite_language)}
```

Comment:
I see what you're trying to do here with fido and fetch, and I agree it's funny, but unfortunately it's also harder to understand. Better would be funny AND helpful.